### ISSUE TITLE
QUnit 2.0.0: Timeout when using promises in hooks

### ISSUE THREAD

**Issue by tbiesemann** [jquery/qunit#1023]
Hi,

The following tests won't be executed:

`
((function(QUnit) {

	var callbackOrder = [];
	QUnit.module("Check whether QUnit handles promises properly", {
		before: function(assert) {
			return new Promise(function(resolve) {
				callbackOrder.push("before");
				resolve();
			});
		},
		beforeEach: function(assert) {
			return new Promise(function(resolve) {
				callbackOrder.push("beforeEach");
				resolve();
			});
		},
		afterEach: function(assert) {
			return new Promise(function(resolve) {
				callbackOrder.push("afterEach");
				resolve();
			});
		},
		after: function(assert) {
			return new Promise(function(resolve) {
				callbackOrder.push("after");
				resolve();
			});
		}
	});

	QUnit.test("Check QUnit callbacks in first test", function(assert) {
		return new Promise(function(resolve) {
			assert.deepEqual(callbackOrder, ["before", "beforeEach"]);
			resolve();
		});
	});

	QUnit.test("Check QUnit callbacks in second test", function(assert) {
		return new Promise(function(resolve) {
			assert.deepEqual(callbackOrder, ["before", "beforeEach", "afterEach", "beforeEach"]);
			resolve();
		});
	});
})(QUnit));
`

I guess QUnit has a problem when dealing with multiple Promises.
It seems to work if only `before` or only `beforeEach` is used.

Thanks, 
Thomas

**Comment by trentmwillis** [comment]
Sounds like a duplicate of https://github.com/jquery/qunit/issues/1016. This has been fixed in master and we'll try to get a release out with it soon.

**Comment by trentmwillis** [comment]
Should be fixed with the new release `2.0.1`
